nebula_decoders/src/nebula_decoders_continental/decoders/continental_ars548_decoder.cpp

```diff
@@ -0,0 +1,574 @@
+
+  radar_status_.longitudinal_velocity_status =
+    sensor_status_packet.longitudinal_velocity_status == 0   ? "VDY_OK"
+    : sensor_status_packet.longitudinal_velocity_status == 1 ? "VDY_NOTOK"
+                                                             : "INVALID VALUE";
+
+  radar_status_.longitudinal_acceleration_status =
+    sensor_status_packet.longitudinal_acceleration_status == 0   ? "VDY_OK"
+    : sensor_status_packet.longitudinal_acceleration_status == 1 ? "VDY_NOTOK"
+                                                                 : "INVALID VALUE";
+
+  radar_status_.lateral_acceleration_status =
+    sensor_status_packet.lateral_acceleration_status == 0   ? "VDY_OK"
+    : sensor_status_packet.lateral_acceleration_status == 1 ? "VDY_NOTOK"
+                                                            : "INVALID VALUE";
+
+  radar_status_.yaw_rate_status = sensor_status_packet.yaw_rate_status == 0   ? "VDY_OK"
+                                  : sensor_status_packet.yaw_rate_status == 1 ? "VDY_NOTOK"
+                                                                              : "INVALID VALUE";
+
+  radar_status_.steering_angle_status = sensor_status_packet.steering_angle_status == 0 ? "VDY_OK"
+                                        : sensor_status_packet.steering_angle_status == 1
+                                          ? "VDY_NOTOK"
+                                          : "INVALID VALUE";
+
+  radar_status_.driving_direction_status =
+    sensor_status_packet.driving_direction_status == 0   ? "VDY_OK"
+    : sensor_status_packet.driving_direction_status == 1 ? "VDY_NOTOK"
+                                                         : "INVALID VALUE";
+
+  radar_status_.characteristic_speed_status =
+    sensor_status_packet.characteristic_speed_status == 0   ? "VDY_OK"
+    : sensor_status_packet.characteristic_speed_status == 1 ? "VDY_NOTOK"
+                                                            : "INVALID VALUE";
+
+  if (sensor_status_packet.radar_status == 0) {
+    radar_status_.radar_status = "STATE_INIT";
+  } else if (sensor_status_packet.radar_status == 1) {
+    radar_status_.radar_status = "STATE_OK";
+  } else if (sensor_status_packet.radar_status == 2) {
+    radar_status_.radar_status = "STATE_INVALID";
+  } else {
+    radar_status_.radar_status = "INVALID VALUE";
+  }
+
+  if (sensor_status_packet.voltage_status == 0) {
+    radar_status_.voltage_status = "Ok";
+  }
+  if (sensor_status_packet.voltage_status & 0x01) {
+    radar_status_.voltage_status += "Current under voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x02) {
+    radar_status_.voltage_status = "Past under voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x03) {
+    radar_status_.voltage_status = "Current over voltage";
+  }
+  if (sensor_status_packet.voltage_status & 0x04) {
+    radar_status_.voltage_status = "Past over voltage";
+  }
```

Comment:
The bitfields for voltage/temperature status are currently not parsed correctly (0x03, 0x04 are not correct for overvoltage/overtemperature; there can be combinations of current/past over/undervoltage).
It should be `0x04, 0x08` respectively.
Also, undervoltage etc. should be spelled as single words.

It would also be nice to have the statuses comma separated in case a combination of them applies but I realize this is a pain in CPP, so I think comma separation with a trailing comma is also fine (or using `boost::algorithm::join(..)`).

(see IO datasheet)